### PR TITLE
Sklearn update

### PR DESCRIPTION
**Pull request by cae67**
**OVERVIEW:**
Updated polyssifier to enable it to install via pip and run in Python 3 while maintaining Python 2.7 compatibility. 

Note that due to differences in Windows and Linux multiprocessing an additional line of code must be added prior to calling polyssifier in Windows:

if __name__ == '__main__':
         report = polyssifier.poly(data,labels)

**TESTING:**
Passed tests on Linux Python 3.7-3.12
Passed tests on Windows Python 3.7-3.12
Passed tests on Windows Python 2.7
Passed most tests on Linux Python 2.7

**MODIFICATIONS:**
Setup.py – changed ‘sklearn’ in install_requires to ‘scikit-learn’
Setup.py – added ‘joblib’ to install_requires to maintain Python 2.7 compatibility
Setup.py – added Python versions 3.7-3.12
Polyssifier.py – line 58 changed because fstring doesn’t work in Python 2
Poly_utils.py – lines 128 and 135 changed because max_features=’auto’ only works in scikit-learn 0.x. Switched to ‘sqrt’, which has identical outcome to ‘auto’ in scikit-learn 0.x but also works in scikit-learn 1.x.
Test_classification.py – lines 39 and 48, changed iteritems() to items() because iteritems() was deprecated in pandas 2.0
Test_multiclass.py – line 29, changed iteritems() to items() because iteritems() was deprecated in pandas 2.0
Test_regression.py – line 28, changed iteritems() to items() because iteritems() was deprecated in pandas 2.0

### CODE REVIEW

**Review by bbradt — APPROVED** [review]
All changes look reasonable to me. If the library works per the current version of the README, I would recommend merging.

File: polyssifier/poly_utils.py
Comment (sergeyplis) [inline polyssifier/poly_utils.py:1]: why did you make that change? why not leave it at `auto`?

**Review by sergeyplis — COMMENTED** [review]
just one comment about `auto`

File: polyssifier/poly_utils.py
Comment (cae67) [reply polyssifier/poly_utils.py:1]: 'auto' was deprecated for tree-based classifiers in more recent versions of Scikit-Learn, so it will cause errors when running. However, in older versions of scikit-learn, 'auto' was identical to 'sqrt', so changing it to 'sqrt' removes errors in newer scikit-learn versions while maintaining compatibility across versions and not modifying how the model is practically trained.

Comment (bbradt) [reply polyssifier/poly_utils.py:1]: Good question... now that I look at the docs, "auto" isn't an option for these model classes...

**Review by sergeyplis — APPROVED** [review]
all's clear
